slick/src/main/scala/slick/jdbc/JdbcDataSource.scala

```diff
@@ -173,6 +178,24 @@ object HikariCPJdbcDataSource extends JdbcDataSourceFactory {
     val ds = new HikariDataSource(hconf)
     new HikariCPJdbcDataSource(ds, hconf)
   }
+
+  def extractUrl(maybeUrl: Option[String]): (Option[String], Option[(String, String)]) = {
+    maybeUrl match {
+      case Some(PostgresFullUrl(username, password, host, dbname)) =>
+        Some(s"jdbc:postgresql://$host/$dbname") -> Some(username -> password)
+
+      case Some(url @ MysqlFullUrl(username, password, host, dbname)) =>
+        val defaultProperties = "?useUnicode=yes&characterEncoding=UTF-8&connectionCollation=utf8_general_ci"
```

Comment:
@cneijenhuis thanks, that clears some things up for me.

Let me see if I have this correct. When play accepted `db.default.url="postgres://..."` it called it's extractUrl function before passing it on to slick. But now, play-slick uses `slick.dbs.default.db.url="..."` which passes all the options under `slick.dbs` straight to slick without manipulating them.

Sound right?